scylladb/sphinx-scylladb-theme
# docs: Python 3.12 compatibility

Closes #1007 

Updates the theme to be compatible with both 3.10 and 3.12 Python versions.

Changes were tested against https://github.com/scylladb/scylladb to ensure it does not introduce breaking changes.

## How to test

1. Go to the `docs` folder.

2. With [pyenv](https://github.com/pyenv/pyenv) (Linux and Mac) or [pyenvwin](https://github.com/pyenv-win/pyenv-win) (Win), install Python 3.12:

    ```sh
    pyenv install 3.12.2
    ```

3. Set Python 3.12 as the default version:

    ```sh
    pyenv local 3.12.2
    ```

4. Set poetry to use Python 3.12:

    ```sh
    poetry env use python
    ```

5. Build the docs:

    ```sh
    make preview
    ```

6. Ensure the build runs without errors.

7. Repeat the process with Python 3.10:

    ```sh
    pyenv install 3.10.11
    pyenv local 3.10.11
    poetry env use python
    make preview
    ```
